ocaml/xapi/test_vdi_allowed_operations.ml

```diff
@@ -294,19 +294,38 @@ let all_cbt_operations = [`enable_cbt; `disable_cbt; `data_destroy] in
 
         (fun vdi -> pass_data_destroy vdi;
           Db.VDI.set_cbt_enabled ~__context ~self:vdi ~value:false
-        ) ,         Some (Api_errors.vdi_no_cbt_metadata, []) ;
+        ) ,         Some (Api_errors.vdi_no_cbt_metadata , []) ;
 
         (fun vdi -> let vM =
                       Test_common.make_vm ~__context () in
           let _: _ API.Ref.t = Test_common.make_vbd ~__context ~vDI:vdi ~vM ~currently_attached:true () in
           pass_data_destroy vdi
-        ) ,         Some (Api_errors.vdi_in_use, []) ;
+        ) ,         Some (Api_errors.vdi_in_use , []) ;
 
         (fun vdi -> pass_data_destroy vdi;
           Db.VDI.set_type ~__context ~self:vdi ~value:`cbt_metadata
         ) ,         None ;
       ] in
 
+  let test_vdi_list_changed_blocks () =
+    let __context = Mock.make_context_with_new_db "Mock context" in
+    List.iter (fun (vdi_fun,api_error) -> run_assert_equal_with_vdi ~__context
+                  ~vdi_fun `list_changed_blocks api_error)
+      [
+        (* check correct error is thrown with live VDI *)
+        ( (fun vDI ->
+              let vM = Test_common.make_vm ~__context () in
+              let _: _ API.Ref.t = Test_common.make_vbd ~__context ~currently_attached:true ~vM ~vDI () in ()
+            ) ,    Some (Api_errors.vdi_in_use , []) ) ;
+        (* check function still runs on CBT metadata/enabled VDIs *)
```

Comment:
Same as below: this *only* works on CBT enabled VDIs in practice, therefore it's best to rephrase / remove this comment for now, because it might confuse people.